packages/core/src/queries/hooks.ts

```diff
@@ -1,23 +1,32 @@
 import type { CreateHook, Hook } from '@logto/schemas';
 import { Hooks } from '@logto/schemas';
-import type { OmitAutoSetFields } from '@logto/shared';
-import { convertToIdentifiers, manyRows } from '@logto/shared';
+import {
+  type OmitAutoSetFields,
+  conditionalSql,
+  convertToIdentifiers,
+  manyRows,
+} from '@logto/shared';
 import type { CommonQueryMethods } from 'slonik';
 import { sql } from 'slonik';
 
 import { buildFindEntityByIdWithPool } from '#src/database/find-entity-by-id.js';
 import { buildInsertIntoWithPool } from '#src/database/insert-into.js';
+import { getTotalRowCountWithPool } from '#src/database/row-count.js';
 import { buildUpdateWhereWithPool } from '#src/database/update-where.js';
 import { DeletionError } from '#src/errors/SlonikError/index.js';
 
 const { table, fields } = convertToIdentifiers(Hooks);
 
 export const createHooksQueries = (pool: CommonQueryMethods) => {
-  const findAllHooks = async () =>
+  const findTotalNumberOfHooks = async () => getTotalRowCountWithPool(pool)(table);
+
+  const findAllHooks = async (limit?: number, offset?: number) =>
     manyRows(
       pool.query<Hook>(sql`
         select ${sql.join(Object.values(fields), sql`, `)}
         from ${table}
+        ${conditionalSql(limit, (limit) => sql`limit ${limit}`)}
+        ${conditionalSql(offset, (offset) => sql`offset ${offset}`)}
```

Comment:
Just curious, are those pagination-related query logics shared by all the resources which has pagination enabled? 
Should we have a more general query factory method implemented, sth like createQueryWithPagination().  
Just did a quick search we have had application and resource two queries sharing a similar logic already, I would prefer to extract this method if it used by more than three places.

Reply:
Updated.